src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xencoder/HaivisionXEncoderCommunicator.java

```diff
@@ -1786,6 +2683,1341 @@ private int getValueByRange(int min, int max, String value) {
+	}
+
+	/**
+	 * Update dropdown list for the control
+	 */
+	private void updateDropdownListForTheControllingMetric() {
+		Map<String, String> stats = localCreateOutputStream.getStatistics();
+		List<AdvancedControllableProperty> advancedControllableProperties = localCreateOutputStream.getControllableProperties();
+		//update still Image
+		String stillImageName = EncoderConstant.CREATE_STREAM + EncoderConstant.HASH + StreamControllingMetric.STILL_IMAGE.getName();
+		String stillImageValue = getEmptyValueForNullData(stats.get(stillImageName));
+		String[] stillImageDropdown = stillImageList.toArray(new String[0]);
+		if (!stillImageList.contains(stillImageValue)) {
+			stillImageValue = EncoderConstant.NONE;
+		}
+		AdvancedControllableProperty stillImageProperty = controlDropdownAcceptNoneValue(stats, stillImageDropdown, stillImageName, stillImageValue);
+		addOrUpdateAdvanceControlProperties(advancedControllableProperties, stillImageProperty);
+
+		String audioSourceName;
+		for (int i = 0; i <= 7; i++) {
+			audioSourceName = EncoderConstant.CREATE_STREAM + EncoderConstant.HASH + StreamControllingMetric.SOURCE_AUDIO.getName() + EncoderConstant.SPACE + i;
+			String propertyName = StreamControllingMetric.SOURCE_AUDIO.getName() + EncoderConstant.SPACE + i;
+			String value = stats.get(audioSourceName);
+			if (!StringUtils.isNullOrEmpty(value)) {
+				for (String key : nameAndAudioConfigMap.keySet()) {
+					if (key.contains(value) || value.contains(key)) {
+						value = key;
+						break;
+					}
+				}
+				updateSourceAudioDropdownList(audioSourceName, value, stats, advancedControllableProperties, propertyName);
+			}
+		}
+	}
+
+	/**
+	 * Add new Source Audio for output stream
+	 *
+	 * @param stats is list of stats
+	 * @param advancedControllablePropertyList is list AdvancedControllableProperty instance
+	 * @throws ResourceNotReachableException if assign more than 7 source audio
+	 */
+	private void addSourceAudioForOutputStream(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllablePropertyList) {
+		nameAndAudioConfigMap.put(EncoderConstant.NONE, new AudioConfig());
+		String prefixName = EncoderConstant.CREATE_STREAM + EncoderConstant.HASH;
+		String[] audioNames = nameAndAudioConfigMap.keySet().toArray(new String[0]);
+		Arrays.sort(audioNames);
+		Map<Integer, String> audioIdList = new HashMap<>();
+		int countSource = 0;
+
+		//Check list source audio assigned in stream
+		for (Map.Entry<String, Audio> audioKey : nameAndSourceAudioMap.entrySet()) {
+			if (audioKey.getValue() != null && audioKey.getValue().getAudioId() != null) {
+				audioIdList.put(Integer.valueOf(audioKey.getValue().getAudioId()), audioKey.getValue().getAudioName());
+				countSource++;
+			}
+		}
+		//Maximum of source audio is 8,
+		if (countSource == EncoderConstant.MAX_SOURCE_AUDIO_DROPDOWN) {
+			throw new ResourceNotReachableException("The audio source just assign max audio source is 8.");
```

Comment:
should it be 8 max but still allowed? In this case, wouldn't it be anything more than 8 instead, before throwing an exception? Or does 8 mean that something went wrong already?